libpod/container_top_linux.go

```diff
@@ -328,7 +328,7 @@ func (c *Container) execPS(psArgs []string) ([]string, bool, error) {
 	cmd.Stdout = wPipe
 	cmd.Stderr = &errBuf
 	// nil means use current env so explicitly unset all, to not leak any sensitive env vars
-	cmd.Env = []string{}
+	cmd.Env = []string{"XDG_CONFIG_HOME=/tmp"}
```

Comment:
Is this necessary? And is it safe? Any unprivileged user account can interfere with the paths being used.

Reply:
we are using it only to re-execute `podman top` and the storage version of `GetConfigHome()` fails if there is no `XDG_CONFIG_HOME` specified.

Reply:
This seems like a big problem then. Shouldn't we instead pass down $HOME then instead of setting something completely invalid. containers.conf will be read from /tmp/containers/containers.conf with this chnage and that means other users can interfere and abuse that fact.

Reply:
I’m unfamiliar with the code and this is a stupid question … why does the in-container subprocess read any configs at all? I don’t think it should read any configuration from inside the container in principle, and I also can’t see any code doing that in `podmanTopInner`.

I don’t doubt that this code is necessary, and it probably trivially shows up in a test, still, what is going on? Is something processing the environment before `reexec` gets to intervene, or something?

Reply:
podman loads the config in init() functions so yes every reexec will end up reading our configs files.

Reply:
I've changed it to just propagate `HOME`

Reply:
`cmd/podman/root.go:init` and `pkg/utils/utils.go:init`? Maybe others?

Could we, instead, allow disabling that behavior, maybe by defining a new `__PODMAN_DO_NOT_INITIALIZE_CONFIG` environment variable?

It seems to me that there is no directory inside the container that is “safe” (at least against `ps` being broken by placing an invalid config); we could mount an empty directory inside the container (or inside some private namespace the `ps` is running in?) but that seems like even more work.

Reply:
I mean this seems totally out of scope of this PR and I don't really see what changed to before were it worked without it?

In any case the proper "fix" is to stop doing things in `init()`, it just results in bad code that will be run regardless if it is needed or not. If we do not use init() then we can have rexxec as first call in main() and only after that start assembling podman config+cli which should also make the rexxec faster.